### ISSUE TITLE
Need clarification for UDP-to-TCP gateway example

### ISSUE THREAD

**Issue by brianraymor** [core-wg/coap-tcp-tls#14]
Capturing embedded issues from #2:

_a UDP-to-TCP gateway may simply not have the context to convert a message with a Block option into the equivalent exchange without any use of a Block option._  

@cabo commented: I don't understand this argument.

**Comment by brianraymor** [comment]
@cabo - who authored that text? Hannes responded that he did not write this.

**Comment by juanjperez** [comment]
I don't know who added this, but I do remember a conversation in Yokohama around the scenario of the UDP-TCP gateway needing to buffer the complete set of blocks (from the UDP side) before passing it onto the TCP side.  For large numbers of blocks, this would result in a large storage requirement on the gateway in order to buffer the entire set of blocks.

**Comment by cabo** [comment]
This text came in slightly before the July 2015 meeting.
I put it in, but that doesn't mean I wrote it...

```
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 387) need for the Block protocol {{-block}}.  It must be noted that
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 388) entirely getting rid of the block protocol is
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 389) not a generally applicable solution, as:
a65bf6a6 draft-tschofenig-core-coap-tcp-tls-xx.mkd (cabo@tzi.org              2015-07-15 390) 
a65bf6a6 draft-tschofenig-core-coap-tcp-tls-xx.mkd (cabo@tzi.org              2015-07-15 391) * a UDP-to-TCP gateway may simply not have the context to convert a
a65bf6a6 draft-tschofenig-core-coap-tcp-tls-xx.mkd (cabo@tzi.org              2015-07-15 392)   message with a Block option into the equivalent exchange without any
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 393)   use of a Block option;
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 394) * large messages might also cause undesired head-of-line blocking;
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 395) * the 2-byte message length field causes another, larger upper bound to the
e4ad0edf draft-ietf-core-coap-tcp-tls-xx.mkd       (cabo@tzi.org              2016-05-25 396)   message length.
```

**Comment by juanjperez** [comment]
I spoke with Elio and he noted that it was a f2f comment at the Yokohama meeting.  Perhaps it's in the meeting minutes.

**Comment by brianraymor** [comment]
Closed in #46
